src/Features/Core/Portable/MakeMemberStatic/AbstractMakeMemberStaticCodeFixProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.MakeMemberStatic
+{
+    internal abstract class AbstractMakeMemberStaticCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.Compile;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            if (context.Diagnostics.Length != 1 ||
+                context.Diagnostics[0].Location?.FindNode(context.CancellationToken) == null)
+            {
+                return Task.CompletedTask;
+            }
+
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected sealed override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor,
+            CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < diagnostics.Length; i++)
+            {
+                var declaration = diagnostics[i].Location?.FindNode(cancellationToken);
+                if (declaration != null)
```

Comment:
this approach is pretty unsafe.  we don't control the diagnostic that is created, so it may be on some random node/sub-node.

Reply:
@CyrusNajmabadi Are you suggesting to add a check to ensure that the node is a member and bail out if it isn't?

Reply:
@mavasani here (cc @CyrusNajmabadi)

Reply:
Yes, I believe the suggestion is to check the node is of expected syntax kind or has an expected base syntax type.